{"name": "test_steps_demo[unittest]", "status": "broken", "statusDetails": {"message": "FileNotFoundError: [Errno 2] No such file or directory: './restlt/b.png'", "trace": "test_data = 'unittest'\n\n    @allure.testcase(\"http://github.com\",\"测试用例管理\")#连接到测试用例管理的一个网页\n    @allure.feature(\"百度搜索\")\n    @pytest.mark.parametrize('test_data',['allure','pytest','unittest'])\n    def test_steps_demo(test_data):\n    \n        with allure.step(\"打开百度网页\"):\n            driver = webdriver.Chrome()\n            driver.get(\"https://www.baidu.com/\")\n    \n        with allure.step(f\"输入搜索词:{'test_data'}\"):\n            driver.find_element_by_id(\"kw\").send_keys(test_data)\n            sleep(2)\n            driver.find_element_by_id(\"su\").click()\n            sleep(2)\n    \n        with allure.step(\"保存图片\"):\n            driver.save_screenshot(\"./result/b.png\")\n>           allure.attach.file(\"./restlt/b.png\",attachment_type=allure.attachment_type.PNG)\n\ntest_baidu.py:25: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\venv\\lib\\site-packages\\allure_commons\\_allure.py:167: in file\n    plugin_manager.hook.attach_file(source=source, name=name, attachment_type=attachment_type, extension=extension)\n..\\venv\\lib\\site-packages\\pluggy\\hooks.py:286: in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n..\\venv\\lib\\site-packages\\pluggy\\manager.py:93: in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n..\\venv\\lib\\site-packages\\pluggy\\manager.py:84: in <lambda>\n    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(\n..\\venv\\lib\\site-packages\\allure_pytest\\listener.py:219: in attach_file\n    self.allure_logger.attach_file(uuid4(), source, name=name, attachment_type=attachment_type, extension=extension)\n..\\venv\\lib\\site-packages\\allure_commons\\reporter.py:113: in attach_file\n    plugin_manager.hook.report_attached_file(source=source, file_name=file_name)\n..\\venv\\lib\\site-packages\\pluggy\\hooks.py:286: in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n..\\venv\\lib\\site-packages\\pluggy\\manager.py:93: in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n..\\venv\\lib\\site-packages\\pluggy\\manager.py:84: in <lambda>\n    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(\n..\\venv\\lib\\site-packages\\allure_commons\\logger.py:53: in report_attached_file\n    shutil.copy2(source, destination)\nC:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\shutil.py:432: in copy2\n    copyfile(src, dst, follow_symlinks=follow_symlinks)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nsrc = './restlt/b.png', dst = 'E:\\\\warmer\\\\ttas\\\\allure_pytest\\\\result\\\\1\\\\03bc5e57-0fef-45b5-847a-35b8bb200a9f-attachment.png'\n\n    def copyfile(src, dst, *, follow_symlinks=True):\n        \"\"\"Copy data from src to dst in the most efficient way possible.\n    \n        If follow_symlinks is not set and src is a symbolic link, a new\n        symlink will be created instead of copying the file it points to.\n    \n        \"\"\"\n        sys.audit(\"shutil.copyfile\", src, dst)\n    \n        if _samefile(src, dst):\n            raise SameFileError(\"{!r} and {!r} are the same file\".format(src, dst))\n    \n        file_size = 0\n        for i, fn in enumerate([src, dst]):\n            try:\n                st = _stat(fn)\n            except OSError:\n                # File most likely does not exist\n                pass\n            else:\n                # XXX What about other special files? (sockets, devices...)\n                if stat.S_ISFIFO(st.st_mode):\n                    fn = fn.path if isinstance(fn, os.DirEntry) else fn\n                    raise SpecialFileError(\"`%s` is a named pipe\" % fn)\n                if _WINDOWS and i == 0:\n                    file_size = st.st_size\n    \n        if not follow_symlinks and _islink(src):\n            os.symlink(os.readlink(src), dst)\n        else:\n>           with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:\nE           FileNotFoundError: [Errno 2] No such file or directory: './restlt/b.png'\n\nC:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\shutil.py:261: FileNotFoundError"}, "steps": [{"name": "打开百度网页", "status": "passed", "start": 1593681549253, "stop": 1593681555462}, {"name": "输入搜索词:test_data", "status": "passed", "start": 1593681555462, "stop": 1593681559737}, {"name": "保存图片", "status": "broken", "statusDetails": {"message": "FileNotFoundError: [Errno 2] No such file or directory: './restlt/b.png'\n", "trace": "  File \"E:\\warmer\\ttas\\allure_pytest\\test_baidu.py\", line 25, in test_steps_demo\n    allure.attach.file(\"./restlt/b.png\",attachment_type=allure.attachment_type.PNG)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\allure_commons\\_allure.py\", line 167, in file\n    plugin_manager.hook.attach_file(source=source, name=name, attachment_type=attachment_type, extension=extension)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\hooks.py\", line 286, in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\manager.py\", line 93, in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\manager.py\", line 84, in <lambda>\n    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\callers.py\", line 208, in _multicall\n    return outcome.get_result()\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\callers.py\", line 80, in get_result\n    raise ex[1].with_traceback(ex[2])\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\callers.py\", line 187, in _multicall\n    res = hook_impl.function(*args)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\allure_pytest\\listener.py\", line 219, in attach_file\n    self.allure_logger.attach_file(uuid4(), source, name=name, attachment_type=attachment_type, extension=extension)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\allure_commons\\reporter.py\", line 113, in attach_file\n    plugin_manager.hook.report_attached_file(source=source, file_name=file_name)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\hooks.py\", line 286, in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\manager.py\", line 93, in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\manager.py\", line 84, in <lambda>\n    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\callers.py\", line 208, in _multicall\n    return outcome.get_result()\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\callers.py\", line 80, in get_result\n    raise ex[1].with_traceback(ex[2])\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\pluggy\\callers.py\", line 187, in _multicall\n    res = hook_impl.function(*args)\n  File \"e:\\warmer\\ttas\\venv\\lib\\site-packages\\allure_commons\\logger.py\", line 53, in report_attached_file\n    shutil.copy2(source, destination)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\shutil.py\", line 432, in copy2\n    copyfile(src, dst, follow_symlinks=follow_symlinks)\n  File \"C:\\Users\\Administrator\\AppData\\Local\\Programs\\Python\\Python38\\lib\\shutil.py\", line 261, in copyfile\n    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:\n"}, "attachments": [{"source": "03bc5e57-0fef-45b5-847a-35b8bb200a9f-attachment.png", "type": "image/png"}], "start": 1593681559737, "stop": 1593681560103}], "parameters": [{"name": "test_data", "value": "'unittest'"}], "start": 1593681549253, "stop": 1593681560104, "uuid": "1faab060-1317-47b6-bc3c-6d1cb35c5ad4", "historyId": "a8b8620c97d46b085352514fdda8f491", "testCaseId": "389a0fa3b7ce74fdf8d144ac6d62347c", "fullName": "test_baidu#test_steps_demo", "labels": [{"name": "feature", "value": "百度搜索"}, {"name": "suite", "value": "test_baidu"}, {"name": "host", "value": "SC-202006280641"}, {"name": "thread", "value": "6180-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test_baidu"}], "links": [{"type": "test_case", "url": "http://github.com", "name": "测试用例管理"}]}